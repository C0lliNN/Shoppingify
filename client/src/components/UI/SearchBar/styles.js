import styled from 'styled-components';
import * as variables from '../../../helpers/style-constants';

export const IconContainer = styled.div`
  background-color: ${variables.COLORS.white};
  box-shadow: 0px 2px 12px rgba(0, 0, 0, 0.04);
  border-top-left-radius: ${variables.BORDER_RADIUS_1};
  border-bottom-left-radius: ${variables.BORDER_RADIUS_1};
  height: 100%;
  box-sizing: border-box;
  display: flex;
  align-items: center;
  padding: 0px 10px;
`;

export const Icon = styled.div`
  color: ${variables.COLORS.black_2};
  font-size: ${variables.FONT_SIZE_4};
`;

export const Input = styled.input`
  background-color: ${variables.COLORS.white};
  box-shadow: 0px 2px 12px rgba(0, 0, 0, 0.04);
  border-top-right-radius: ${variables.BORDER_RADIUS_1};
  border-bottom-right-radius: ${variables.BORDER_RADIUS_1};
  font-family: ${variables.FONT_FAMILY};
  font-weight: 600;
  height: 100%;
  border: none;
  outline: none;
  &::placeholder {
    color: ${variables.COLORS.gray_2};
  }
`;

export const StyledSearchBar = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  height: 40px;
  margin-top: 10px;
  @media (min-width: ${variables.LG_BREAK_POINT}px) {
    font-size: ${variables.FONT_SIZE_4};
    margin-left: 20px;
    margin-top: 0px;
  }
`;