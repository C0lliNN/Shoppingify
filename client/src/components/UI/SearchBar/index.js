import React from 'react';
import { StyledSearchBar, IconContainer, Icon, Input } from './styles';

function SearchBar() {
  return (
    <StyledSearchBar>
      <IconContainer>
        <Icon className="material-icons-round">search</Icon>
      </IconContainer>
      <Input placeholder="Search item" />
    </StyledSearchBar>
  );
}

export default SearchBar;
