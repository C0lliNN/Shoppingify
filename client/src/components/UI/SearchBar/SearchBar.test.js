import React from 'react';
import { render } from '@testing-library/react';
import SearchBar from './SearchBar';

describe('<SearchBar/>', () => {
  it('should render correctly', () => {
    const { container } = render(<SearchBar />);
    expect(container).toMatchSnapshot();
  });
});
