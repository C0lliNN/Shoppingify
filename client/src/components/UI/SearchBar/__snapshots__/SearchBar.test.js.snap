// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar/> should render correctly 1`] = `
<div>
  <div
    class="sc-AxhCb cATiUb"
  >
    <div
      class="sc-AxjAm iBAYIb"
    >
      <div
        class="sc-AxirZ jQSkBK material-icons-round"
      >
        search
      </div>
    </div>
    <input
      class="sc-AxiKw hGySNu"
      placeholder="Search item"
    />
  </div>
</div>
`;
